modules/openthread/platform/udp.c

```diff
@@ -0,0 +1,452 @@
+/*
+ * Copyright 2025 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "openthread/udp.h"
+#include "openthread/ip6.h"
+#include "openthread/platform/udp.h"
+#include "openthread_border_router.h"
+#include "sockets_internal.h"
+#include <common/code_utils.hpp>
+#include <errno.h>
+#include <openthread.h>
+#include <openthread/logging.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <zephyr/kernel.h>
+#include <zephyr/net/net_if.h>
+#include <zephyr/net/net_ip.h>
+#include <zephyr/net/socket.h>
+#include <zephyr/net/socket_service.h>
+#include <zephyr/posix/arpa/inet.h>
+#include <zephyr/posix/unistd.h>
+#include <zephyr/sys/util.h>
+
+static struct zsock_pollfd sockfd_udp[CONFIG_OPENTHREAD_ZEPHYR_BORDER_ROUTER_MAX_UDP_SERVICES];
+
+static struct otInstance *ot_instance_ptr;
+static struct net_if *ot_iface_ptr, *ail_iface_ptr;
+static uint32_t ot_iface_index, ail_iface_index;
+
+static void udp_receive_handler(struct net_socket_service_event *evt);
+
+NET_SOCKET_SERVICE_SYNC_DEFINE_STATIC(handle_udp_receive, udp_receive_handler,
+				      CONFIG_OPENTHREAD_ZEPHYR_BORDER_ROUTER_MAX_UDP_SERVICES);
+
+otError udp_plat_init(otInstance *ot_instance, struct net_if *ail_iface, struct net_if *ot_iface)
+{
+	ot_instance_ptr = ot_instance;
+	ot_iface_ptr = ot_iface;
+	ail_iface_ptr = ail_iface;
+	ot_iface_index = (uint32_t)net_if_get_by_iface(ot_iface);
+	ail_iface_index = (uint32_t)net_if_get_by_iface(ail_iface);
+
+	for (uint8_t i = 0; i < CONFIG_OPENTHREAD_ZEPHYR_BORDER_ROUTER_MAX_UDP_SERVICES; i++) {
+		sockfd_udp[i].fd = -1;
+	}
+
+	return OT_ERROR_NONE;
+}
+
+void udp_plat_deinit(void)
+{
+	ARRAY_FOR_EACH(sockfd_udp, idx) {
+		if (sockfd_udp[idx].fd != -1) {
+			zsock_close(sockfd_udp[idx].fd);
+			sockfd_udp[idx].fd = -1;
+		}
+	}
+}
+
+otError otPlatUdpSocket(otUdpSocket *aUdpSocket)
+{
+	otError error = OT_ERROR_NONE;
+	int sock = -1;
+	static uint8_t sock_cnt;
+
+	VerifyOrExit(aUdpSocket, error = OT_ERROR_INVALID_ARGS);
```

Comment:
No implicit boolean conversions (same for others)

```suggestion
	VerifyOrExit(aUdpSocket != NULL, error = OT_ERROR_INVALID_ARGS);
```